<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>高级串口调试工具</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="configGroup">
      <property name="title">
       <string>串口配置</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="portLabel">
         <property name="text">
          <string>串口号:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="portCombo"/>
       </item>
       <item>
        <widget class="QPushButton" name="refreshBtn">
         <property name="text">
          <string>刷新</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="baudLabel">
         <property name="text">
          <string>波特率:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="baudCombo"/>
       </item>
       <item>
        <widget class="QLabel" name="dataBitsLabel">
         <property name="text">
          <string>数据位:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="dataBitsCombo"/>
       </item>
       <item>
        <widget class="QLabel" name="stopBitsLabel">
         <property name="text">
          <string>停止位:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="stopBitsCombo"/>
       </item>
       <item>
        <widget class="QLabel" name="parityLabel">
         <property name="text">
          <string>校验位:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="parityCombo"/>
       </item>
       <item>
        <widget class="QPushButton" name="openBtn">
         <property name="text">
          <string>打开串口</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QSplitter" name="splitter" orientation="1">
      <property name="childrenCollapsible">
       <bool>false</bool>
      </property>
      <widget class="QGroupBox" name="receiveGroup">
       <property name="title">
        <string>接收数据</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QTextEdit" name="receiveText">
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="lineWrapMode">
           <enum>QTextEdit::WidgetWidth</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QCheckBox" name="timestampCheck">
            <property name="text">
             <string>显示时间戳</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="timeFormatCombo"/>
          </item>
          <item>
           <widget class="QCheckBox" name="hexDisplayCheck">
            <property name="text">
             <string>十六进制显示</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="autoScrollCheck">
            <property name="text">
             <string>自动滚动</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="clearReceiveBtn">
            <property name="text">
             <string>清空接收区</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="saveReceiveBtn">
            <property name="text">
             <string>保存到文件</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="sendGroup">
       <property name="title">
        <string>发送数据</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QTextEdit" name="sendText">
          <property name="lineWrapMode">
           <enum>QTextEdit::WidgetWidth</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QPushButton" name="sendBtn">
            <property name="text">
             <string>发送</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="hexSendCheck">
            <property name="text">
             <string>十六进制发送</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="appendNewlineCheck">
            <property name="text">
             <string>自动添加换行</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="formatJsonBtn">
            <property name="text">
             <string>格式化JSON</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="historyGroup">
       <property name="title">
        <string>发送历史记录</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QListWidget" name="historyList">
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QPushButton" name="resendBtn">
            <property name="text">
             <string>重新发送</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="deleteBtn">
            <property name="text">
             <string>删除记录</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="clearHistoryBtn">
            <property name="text">
             <string>清空历史</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>